bindings/kafka/kafka.go

```diff
@@ -15,321 +15,86 @@ package kafka
 
 import (
```

Comment:
I propose we split into v1 (old code) and v2 (this version). And we invest in making v2 stable and give users time to migrate from v1 to v2 before deprecating.

Can you change one of the apps in E2E tests to explain the breaking changes as per https://github.com/dapr/dapr/pull/4565 ?

Reply:
@skyao if there any breaking changes to the spec, please point them out clearly and answer whether or not we can change this to not have breaking changes.

If these breaking changes (in either .spec.metadata scheme or request/response) cannot be made non-breaking, this must be a v2.

Reply:
From the difference  found, it seems that it's possible to complete this refactory without introduce v2 api. 

Of cause we need to resolve these problems first:

1. pub/sub specify BOTH authRequired: false and disableTLS: true
2. risk of retry mechanism and app return fail for specified message. The API is not changed but the behavior after retry is not compatible.